app/external_doc.rb

```diff
@@ -1,13 +1,121 @@
+require 'html/pipeline'
+require 'uri'
+
 class ExternalDoc
-  def self.fetch(url_to_markdown)
-    markdown = HTTP.get(url_to_markdown)
+  def self.fetch(repository:, path:)
+    contents = HTTP.get(
+      "https://raw.githubusercontent.com/#{repository}/master/#{path}",
+    )
+
+    context = {
+      base_url: URI.join(
+        'https://github.com',
+        "#{repository}/blob/master/",
+      ),
+
+      image_base_url: URI.join(
+        'https://raw.githubusercontent.com',
+        "#{repository}/master/",
+      ),
+    }
+
+    context[:subpage_url] =
+      URI.join(context[:base_url], File.join('.', File.dirname(path), '/'))
+
+    context[:image_subpage_url] =
+      URI.join(context[:image_base_url], File.join('.', File.dirname(path), '/'))
+
+    filters = [
+      HTML::Pipeline::MarkdownFilter,
+      HTML::Pipeline::AbsoluteSourceFilter,
+      PrimaryHeadingFilter,
+      TableOfContentsFilter,
+      AbsoluteLinkFilter,
+      MarkdownLinkFilter,
+    ]
+
+    HTML::Pipeline
+      .new(filters)
+      .to_html(contents, context)
+  end
+
+private
+
+  class AbsoluteLinkFilter < HTML::Pipeline::Filter
+    def call
+      doc.search('a').each do |element|
+        next if element['href'].nil? || element['href'].empty?
+
+        href = element['href'].strip
+        uri = URI.parse(href)
+        path = uri.path
+
+        unless uri.scheme || href.start_with?('#') || path.end_with?('.md')
+          base = if path.start_with? '/'
+                   base_url
+                 else
+                   subpage_url
+                 end
+
+          element['href'] = URI.join(base, href).to_s
+        end
+      end
+
+      doc
+    end
+
+    def subpage_url
+      context[:subpage_url]
+    end
+  end
+
+  class MarkdownLinkFilter < HTML::Pipeline::Filter
+    def call
+      doc.search('a').each do |element|
+        next if element['href'].nil? || element['href'].empty?
+
+        href = element['href'].strip
+        uri = URI.parse(href)
+
+        if uri.path.end_with?('.md')
+          uri.path.sub!(/.md$/, '.html')
+          element['href'] = uri.to_s
+        end
+      end
+
+      doc
+    end
+  end
+
+  class PrimaryHeadingFilter < HTML::Pipeline::Filter
+    def call
+      doc.at('h1:first-of-type').unlink
+      doc
+    end
+  end
+
+  class TableOfContentsFilter < HTML::Pipeline::Filter
```

Comment:
I'm not sure what this is doing, but is this re-implementing the anchored-heading stuff from the docs? I think that's done with JS so this might not be necessary.